src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/PersonPreviousName.java

```diff
@@ -0,0 +1,89 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.Split;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Example PNA+PER++++SU:PATTERSON'
+ */
+
+@Getter
+@Builder
+@EqualsAndHashCode(callSuper = false)
+@ToString
+@Slf4j
+public class PersonPreviousName extends Segment {
+
+    public static final String KEY = "PNA";
+    public static final String QUALIFIER = "PER";
+    public static final String KEY_QUALIFIER = KEY + PLUS_SEPARATOR + QUALIFIER;
+
+    @NonNull
+    private final String previousFamilyName;
+
+    public static PersonPreviousName fromString(String edifactString) {
+        if (!edifactString.startsWith(PersonPreviousName.KEY_QUALIFIER)) {
+            throw new IllegalArgumentException("Can't create " + PersonPreviousName.class.getSimpleName() + " from " + edifactString);
+        }
+
+        var fields = Split.byPlus(edifactString);
+        var builder = PersonPreviousName.builder();
+        if (fields.length >= 6) {
```

Comment:
shouldn't it throw exception when it's shorter than `6`? Can this condition be extracted to method and named?

Reply:
Not explicitly, but it will throw exception on `build()` https://github.com/nhsconnect/integration-adaptor-nhais/pull/147/files/58c9459fffd228c12894a7e07ac33e73ef919300#diff-241f7796852f2b1cfc264d2a33131056R52 because previous name is annotated with `@NonNull` here https://github.com/nhsconnect/integration-adaptor-nhais/pull/147/files/58c9459fffd228c12894a7e07ac33e73ef919300#diff-241f7796852f2b1cfc264d2a33131056R34 and unit test covers that